Patch to fix support for beep-media-player
* BMP (old and new) and XMMS (old) does not require gdk-pixbuf; XMMS (new) however does.
* Fail when errors are detected rather than silently do what we can do.
* Make it possible to build BMP new/old and XMMS new/old independently

diff -Naur xosd-2.2.14.orig/aclocal.m4 xosd-2.2.14/aclocal.m4
--- xosd-2.2.14.orig/aclocal.m4	2005-03-28 02:57:08.000000000 +0200
+++ xosd-2.2.14/aclocal.m4	2005-03-28 04:20:12.000000000 +0200
@@ -1,5 +1,4 @@
-# generated automatically by aclocal 1.9.2 -*- Autoconf -*-
-
+# generated automatically by aclocal 1.9.5 -*- Autoconf -*-
 # Copyright (C) 1996, 1997, 1998, 1999, 2000, 2001, 2002, 2003, 2004
 # Free Software Foundation, Inc.
 # This file is free software; the Free Software Foundation
@@ -6534,7 +6533,7 @@
 # Call AM_AUTOMAKE_VERSION so it can be traced.
 # This function is AC_REQUIREd by AC_INIT_AUTOMAKE.
 AC_DEFUN([AM_SET_CURRENT_AUTOMAKE_VERSION],
-	 [AM_AUTOMAKE_VERSION([1.9.2])])
+	 [AM_AUTOMAKE_VERSION([1.9.5])])
 
 # AM_AUX_DIR_EXPAND
 
diff -Naur xosd-2.2.14.orig/configure.ac xosd-2.2.14/configure.ac
--- xosd-2.2.14.orig/configure.ac	2004-11-17 20:29:58.000000000 +0100
+++ xosd-2.2.14/configure.ac	2005-03-28 16:46:20.000000000 +0200
@@ -22,9 +22,9 @@
 dnl    LT_REVISION = 0;
 dnl    LT_CURRENT ++;
 dnl    if (any interfaces have been _added_ since last release)
-dnl       AGE ++;
+dnl		  AGE ++;
 dnl    if (any interfaces have been _removed_ or _incompatibly changed_)
-dnl       AGE = 0;
+dnl		  AGE = 0;
 dnl }
 LT_CURRENT=4
 LT_AGE=2
@@ -43,7 +43,7 @@
 AM_PROG_LIBTOOL
 AC_PATH_XTRA([X11],[X11/Xlib.h],[XrmInitialize()])
 # We need two libtools, one that builds both shared and static, and
-# one that builds only static.  This is because the resulting libtool
+# one that builds only static.	This is because the resulting libtool
 # does not allow us to choose which to build at runtime.
 sed -e 's/^build_old_libs=yes/build_old_libs=no/' libtool > libtool-disable-static
 chmod +x libtool-disable-static
@@ -55,13 +55,13 @@
 
 dnl Checks for libraries.
 AM_PATH_GTK(1.2.2,,
-	    AC_MSG_WARN([*** GTK+ >= 1.2.2 not found ***]))
+		AC_MSG_WARN([*** GTK+ >= 1.2.2 not found ***]))
 AC_CHECK_LIB(X11, XInitThreads,,
-	     AC_MSG_ERROR([*** X11 not threadsafe ***]))
+		 AC_MSG_ERROR([*** X11 not threadsafe ***]))
 AC_CHECK_LIB(Xext, XShapeQueryExtension,,
-	     AC_MSG_ERROR([*** X11 Shape extension not found ***]))
+		 AC_MSG_ERROR([*** X11 Shape extension not found ***]))
 AC_CHECK_LIB(pthread, pthread_create,,
-	     AC_MSG_ERROR([*** POSIX thread support not found ***]))
+		 AC_MSG_ERROR([*** POSIX thread support not found ***]))
 
 dnl Check for header files.
 AC_HEADER_STDC
@@ -76,101 +76,149 @@
 
 # nicked from the irmix project...
 AC_ARG_ENABLE([xinerama],
-              AC_HELP_STRING([--disable-xinerama],
-			     [disable use of Xinerama extension]),
-              [disable_xinerama="yes"],
-	      [disable_xinerama="no"])
+			  AC_HELP_STRING([--disable-xinerama],
+				 [disable use of Xinerama extension]),
+			  [disable_xinerama="yes"],
+		  [disable_xinerama="no"])
 
 if test x$disable_xinerama = "xno"
 then
-        AC_CHECK_LIB(Xinerama_pic,
-                     XineramaQueryExtension,
-                     [X_LIBS="$X_LIBS -lXinerama_pic"
-                      AC_DEFINE(HAVE_XINERAMA,1,[Define this if you have libXinerama installed])
-                      ac_have_xinerama="yes"],
-                      AC_CHECK_LIB(Xinerama,
-                                   XineramaQueryExtension,
-                                   [X_LIBS="$X_LIBS -lXinerama"
-                                    AC_DEFINE(HAVE_XINERAMA,1,[Define this if you have libXinerama
+		AC_CHECK_LIB(Xinerama_pic,
+					 XineramaQueryExtension,
+					 [X_LIBS="$X_LIBS -lXinerama_pic"
+					  AC_DEFINE(HAVE_XINERAMA,1,[Define this if you have libXinerama installed])
+					  ac_have_xinerama="yes"],
+					  AC_CHECK_LIB(Xinerama,
+								   XineramaQueryExtension,
+								   [X_LIBS="$X_LIBS -lXinerama"
+									AC_DEFINE(HAVE_XINERAMA,1,[Define this if you have libXinerama
 installed])
-                                    ac_have_xinerama="yes"],,
-                                    [$X_LIBS -lXext $X_EXTRA_LIBS]),
-                     [$X_LIBS -lXext $X_EXTRA_LIBS])
+									ac_have_xinerama="yes"],,
+									[$X_LIBS -lXext $X_EXTRA_LIBS]),
+					 [$X_LIBS -lXext $X_EXTRA_LIBS])
 fi
 
+dnl Check if beep-media-player is available
 if pkg-config --exists bmp
 then
 	PKG_CHECK_MODULES(BMP, bmp)
 	BMP_GENERAL_PLUGIN_DIR=$(pkg-config --variable general_plugin_dir bmp)
 	AC_SUBST(BMP_GENERAL_PLUGIN_DIR)
-	no_bmp="yes"
+	no_bmp="no"
 else
-	no_bmp="no"	
+	no_bmp="yes"
 	AC_MSG_WARN("beep media player not found")
 fi
 
-
-
-
 dnl Check for xmms
 AM_PATH_XMMS(1.2.7,,
-	     AC_MSG_WARN("xmms plugins can not be built"))
+		 AC_MSG_WARN("xmms plugins can not be built"))
+
 dnl Override where to place libxmms_xosd.so
 AC_ARG_WITH([plugindir],
-            AC_HELP_STRING([--with-plugindir=DIR],
+			AC_HELP_STRING([--with-plugindir=DIR],
 			   [Set the xmms plugin directory]),
-            [XMMS_PLUGIN_DIR="$withval"],
-            [XMMS_PLUGIN_DIR="$XMMS_GENERAL_PLUGIN_DIR"])
+			[XMMS_PLUGIN_DIR="$withval"],
+			[XMMS_PLUGIN_DIR="$XMMS_GENERAL_PLUGIN_DIR"])
 
 dnl Check for gdk-pixbuf
 AM_PATH_GDK_PIXBUF(0.22.0,,
 		   AC_MSG_WARN("new xmms plugin can not be built"))
+
 dnl Override where to place pixmaps for libxmms_xosd.so
 AC_ARG_WITH([pixmapdir],
-	    AC_HELP_STRING([--with-pixmapdir=DIR],
-			   [Set the directory for xmms plugin pixmaps]),
-            [XMMS_PIXMAPDIR="$withval"],
-            [XMMS_PIXMAPDIR="${datadir}/xosd"])
+		AC_HELP_STRING([--with-pixmapdir=DIR],
+				[Set the directory for xmms plugin pixmaps]),
+			[XMMS_PIXMAPDIR="$withval"],
+		[XMMS_PIXMAPDIR="${datadir}/xosd"])
+
 AC_SUBST(XMMS_PIXMAPDIR)
 
-dnl NEW plugin
-AC_ARG_ENABLE([new-plugin],
-	      AC_HELP_STRING([--disable-new-plugin],
-			     [Disable new xmms plugin (enabled by default)]),
-              [enable_new_plugin="$enableval"],
-	      [enable_new_plugin="yes"])
+dnl XMMS PLUGINS
+dnl new xmms
+AC_ARG_ENABLE([new-xmms],
+		  AC_HELP_STRING([--disable-new-xmms],
+				 [Disable new xmms plugin (enabled by default)]),
+			[enable_new_xmms="$enableval"],
+		  [enable_new_xmms="yes"])
+
+dnl old xmms
+AC_ARG_ENABLE([old-xmms],
+		  AC_HELP_STRING([--enable-old-xmms],
+				[Enable old xmms plugin (disabled by default)]),
+			[enable_old_xmms="$enableval"],
+		  [enable_old_xmms="no"])
+
+
+dnl BEEP-MEDIA-PLAYER PLUGINS
+dnl new bmp
+AC_ARG_ENABLE([new-bmp],
+		AC_HELP_STRING([--disable-new-bmp],
+				[Disable new beep media plugin (enabled by default)]),
+			[enable_new_bmp="$enableval"],
+		  [enable_new_bmp="yes"])
+dnl old bmp
+AC_ARG_ENABLE([old-bmp],
+		  AC_HELP_STRING([--enable-old-bmp],
+				[Enable old bmp plugin (disabled by default)]),
+			[enable_old_bmp="$enableval"],
+		  [enable_old_bmp="no"])
+
+dnl CHECK PLUGINS
+dnl beep media player
+dnl new
+AC_MSG_CHECKING([whether new beep media plugin was requested])
+if test x"$enable_new_bmp" == "xyes"
+then
+	test x"$no_bmp" == "xyes" && AC_MSG_ERROR([yes, but cannot be built])
+	AC_MSG_RESULT([yes])
+else
+	AC_MSG_RESULT([no])
+fi
+AM_CONDITIONAL([BUILD_BEEP_NEW_PLUGIN],
+		   [test x"$enable_new_bmp" = "xyes"])
+dnl old
+AC_MSG_CHECKING([whether old beep media plugin was requested])
+if test x"$enable_old_bmp" == "xyes"
+then
+	test x"$no_bmp" == "xyes" && AC_MSG_ERROR([yes, but cannot be built])
+	AC_MSG_RESULT([yes])
+else
+	AC_MSG_RESULT([no])
+fi
+AM_CONDITIONAL([BUILD_BEEP_OLD_PLUGIN],
+		   [test x"$enable_old_bmp" = "xyes"])
+
+dnl XMMS 
+dnl new 
 AC_MSG_CHECKING([whether new xmms plugin was requested])
-AC_MSG_RESULT($enable_new_plugin)
+if test x"$enable_new_xmms" == "xyes"
+then
+	if test x"$no_xmms" == "xyes" -o x"$no_gdk_pixbuf" == "xyes"
+	then
+		AC_MSG_ERROR([yes, but cannot be built])
+	fi
+	AC_MSG_RESULT([yes])
+else
+	AC_MSG_RESULT([no])
+fi
 AM_CONDITIONAL([BUILD_NEW_PLUGIN],
-	       [test x"$enable_new_plugin" = "xyes" -a x"$no_xmms" != "xyes" -a x"$no_gdk_pixbuf" != "xyes"])
+		   [test x"$enable_new_xmms" = "xyes"])
 
-
-dnl Check for Beep Media player
-AC_ARG_ENABLE([beep_media_player_plugin],
-		AC_HELP_STRING([--enable-beep_media_player_plugin],
-			     [Enable beep media plugin (enabled by default)]),
-              [beep_media_player_plugin="$enableval"],
-              [beep_media_player_plugin="yes"])
-AC_MSG_CHECKING([whether beep media plugin was requested])
-AC_MSG_RESULT($beep_media_player_plugin)
-
-AM_CONDITIONAL([BUILD_BEEP_MEDIA_PLUGIN],
-	       [test x"$beep_media_player_plugin" = "xyes" -a x"$no_bmp" == "xyes"])
-
-		
-
-
-dnl OLD plugin
-AC_ARG_ENABLE([old-plugin],
-	      AC_HELP_STRING([--enable-old-plugin],
-			     [Enable old xmms plugin (disabled by default)]),
-              [enable_old_plugin="$enableval"],
-              [enable_old_plugin="no"])
+dnl old
 AC_MSG_CHECKING([whether old xmms plugin was requested])
-AC_MSG_RESULT($enable_old_plugin)
+if test x"$enable_old_xmms" = "xyes"
+then
+	if test x"$no_xmms" == "xyes"
+	then
+		AC_MSG_ERROR([yes, but cannot be built])
+	fi
+	AC_MSG_RESULT([yes])
+else
+	AC_MSG_RESULT([no])
+fi
 AM_CONDITIONAL([BUILD_OLD_PLUGIN],
-	       [test x"$enable_old_plugin" = "xyes" -a x"$no_xmms" != "xyes"])
-
+		   [test x"$enable_old_xmms" = "xyes"])
 
 dnl Define XOSD_VERSION
 AC_DEFINE_UNQUOTED(XOSD_VERSION, "${VERSION}")
diff -Naur xosd-2.2.14.orig/src/bmp_plugin/Makefile.am xosd-2.2.14/src/bmp_plugin/Makefile.am
--- xosd-2.2.14.orig/src/bmp_plugin/Makefile.am	2005-03-28 16:56:28.000000000 +0200
+++ xosd-2.2.14/src/bmp_plugin/Makefile.am	2005-03-28 16:56:50.000000000 +0200
@@ -7,14 +7,12 @@
 # Don't build a static library
 LIBTOOL = $(top_builddir)/libtool-disable-static
 
-if BUILD_BEEP_MEDIA_PLUGIN
-if BUILD_NEW_PLUGIN
+if BUILD_BEEP_NEW_PLUGIN
 NEW_bmpplugin = libbmp_osd.la
 endif
-if BUILD_OLD_PLUGIN
+if BUILD_BEEP_OLD_PLUGIN
 OLD_bmpplugin = libbmp_osd_old.la
 endif
-endif
 
 bmpplugin_LTLIBRARIES = $(NEW_bmpplugin) $(OLD_bmpplugin)
 
diff -Naur xosd-2.2.14.orig/src/bmp_plugin/bmp_osd.c xosd-2.2.14/src/bmp_plugin/bmp_osd.c
--- xosd-2.2.14.orig/src/bmp_plugin/bmp_osd.c	2004-11-17 20:26:56.000000000 +0100
+++ xosd-2.2.14/src/bmp_plugin/bmp_osd.c	2005-03-28 17:53:27.000000000 +0200
@@ -22,9 +22,9 @@
 
 #include "bmp_osd.h"
 
-#include <xmms/plugin.h>
-#include <xmms/xmmsctrl.h>
-#include <xmms/configfile.h>
+#include <bmp/plugin.h>
+#include <bmp/beepctrl.h>
+#include <bmp/configfile.h>
 
 static void init(void);
 static void cleanup(void);
diff -Naur xosd-2.2.14.orig/src/bmp_plugin/dlg_config.c xosd-2.2.14/src/bmp_plugin/dlg_config.c
--- xosd-2.2.14.orig/src/bmp_plugin/dlg_config.c	2004-11-08 21:03:23.000000000 +0100
+++ xosd-2.2.14/src/bmp_plugin/dlg_config.c	2005-03-28 17:53:27.000000000 +0200
@@ -22,7 +22,7 @@
 
 #include "bmp_osd.h"
 
-#include <xmms/configfile.h>
+#include <bmp/configfile.h>
 
 static GtkToggleButton
   * vol_on, *bal_on,
diff -Naur xosd-2.2.14.orig/src/bmp_plugin/dlg_config_old.c xosd-2.2.14/src/bmp_plugin/dlg_config_old.c
--- xosd-2.2.14.orig/src/bmp_plugin/dlg_config_old.c	2004-11-08 21:03:23.000000000 +0100
+++ xosd-2.2.14/src/bmp_plugin/dlg_config_old.c	2005-03-28 17:53:27.000000000 +0200
@@ -21,7 +21,7 @@
 
 #include "bmp_osd.h"
 
-#include <xmms/configfile.h>
+#include <bmp/configfile.h>
 
 static GtkToggleButton
   * vol_on, *bal_on,
